Fixes #1619 - Adding text boxes for freely inserting random mods on certain vaal notables

Wires77 opened the pull request:
Fixes #1619

ppoelzl reviewed (commented):
When I either
* deselect a selected node or
* select an unselected node

all modified nodes lose their modifed mods, without warning.
This makes it impossible to change your node selection after you put in the modified mods.

Also it seems like Path of Building tries to interpret the "Right click to set mod" text as a mod. As a consequence, after you set a mod, there is no indication that you can change it by right-clicking again. Maybe put the hint in a separate section of the tooltip instead to fix both these issues?

The "Replace modifier of node" dialogue is easy to use. It just seems to be a bit too wide given it's contents.

That said, this is already impressive in its first iteration.

Wires77 commented:
> 
> 
> When I either
> 
>     * deselect a selected node or
> 
>     * select an unselected node
> 
> 
> all modified nodes lose their modifed mods, without warning.
> This makes it impossible to change your node selection after you put in the modified mods.

I can't reproduce this.  This was an old bug I fixed a couple versions ago, though, so if you can consistently reproduce it, let me know.

> Also it seems like Path of Building tries to interpret the "Right click to set mod" text as a mod. As a consequence, after you set a mod, there is no indication that you can change it by right-clicking again. Maybe put the hint in a separate section of the tooltip instead to fix both these issues?

That's a good idea, added to a section close to the mods

> The "Replace modifier of node" dialogue is easy to use. It just seems to be a bit too wide given it's contents.

It actually has to be this wide for some of the mods that can appear on "Militant Faith".  I could potentially make the size dynamic, but I didn't want to over-engineer things.

> That said, this is already impressive in its first iteration.

Most of this has been in master for a couple months, actually.  This PR was mainly to add those two vaal notables, but thanks for the suggestions!

ppoelzl reviewed (approved):
I can't reproduce the issues I had anymore, so this is good to be merged.
